### PR TITLE
[ADD] default_supplierinfo_discount on res.partner model

### PR DESCRIPTION
**Pull request by legalsylvain**
* Add default_supplierinfo_discount field on res.partner.
* this field will be used as the default value, when creating new supplierinfo line for the given supplier.

Misc : Add / improve description.

Do : #298

CC : @JonathanNEMRY, @pedrobaeza. 

Thanks for your review.

**Note** : @pedrobaeza, I tested the code (https://github.com/OCA/partner-contact/blob/9.0/partner_default_sale_discount/models/sale_order.py#L18) but I finally managed it with a simple onchange function. What to you think about that ? It seems to be that it is inheritable too.

reference of your comment : https://github.com/OCA/purchase-workflow/issues/298#issuecomment-259359443

### CODE REVIEW

**Review by JonathanNEMRY — APPROVED** [review]
Thanks for the work!
Code review :+1: 
Juste one point: could you make a litle test for the `onchange_name` method? As it is `onchange` I will not blocking this PR for it but as we already have the tests folder with existing tests inside it could be easy to improve it. thanks in advance! :smile: 
Jonathan

File: product_supplierinfo_discount/models/product_supplierinfo.py [hunk 1]
```diff
@@ -12,3 +13,11 @@ class ProductSupplierInfo(models.Model):
 
     discount = fields.Float(
         string='Discount (%)', digits_compute=dp.get_precision('Discount'))
+
+    @api.onchange('name')
+    @api.multi
+    def onchange_name(self):
+        for supplierinfo in self:
```
Comment (pedrobaeza) [inline product_supplierinfo_discount/models/product_supplierinfo.py]: Do directly:

```
for supplierinfo in self.filtered('name'):
```

and you save one indentation level.

**Review by pedrobaeza — APPROVED** [review]
Little change. This case allows the onchange, as you are not copying previously the value to the product form, so it's good with this indeed.

File: product_supplierinfo_discount/models/product_supplierinfo.py (hunk 1 above)
Comment (legalsylvain) [reply product_supplierinfo_discount/models/product_supplierinfo.py]: Done.

**Comment by legalsylvain** [comment]
@JonathanNEMRY : test added. thanks for your review.

Note : I squashed the commit to make more easy futur git cherry-pick against OCA / 10 branch.

**Review by susport — APPROVED** [review]
👍

**Review by JonathanNEMRY — APPROVED** [review]

**Comment by legalsylvain** [comment]
self merge is not a good practice, but it seems that this PR is ready now.

kind regards.

**Comment by pedrobaeza** [comment]
No need for that, although coming from a trusty source like yours, it can be allowed without problems :wink:

**Pull request merged**